### PR TITLE
Support Rails 7

### LINKED ISSUE #92: Any plans supporting Rails 7?
Hi,

First of all thanks to everyone here for developing this amazing gem.
Noticed that current version is restricted by `'actionpack', '>= 4.2', '< 6.2'`
Any plans to support new Rails 7

Cheers

### PR DESCRIPTION
**Pull request by Nkadze**
This PR add rails 7 support, including ruby 3.0 & 3.1, fixes #92 

It also removes `minitest-rails` from `gemspec` as development dependency and adds it per each `gemfile`, to avoid duplicated or [conditional definitions in gemspec](https://rubystyle.guide/#no-ruby-version-in-the-gemspec)

for rails 7 support it uses branch from [minitest-rails mauricebolhuis fork](https://github.com/mauricebolhuis/minitest-rails/tree/feature/add-rails-7.0.1-support), which hopefully [would be merged](https://github.com/blowmage/minitest-rails/pull/236) soon

### CODE REVIEW

File: stronger_parameters.gemspec [hunk 1]
```diff
@@ -14,14 +14,13 @@ Gem::Specification.new do |spec|
...
   spec.add_development_dependency 'rake'
-  spec.add_development_dependency 'minitest-rails'
   spec.add_development_dependency 'maxitest'
   spec.add_development_dependency 'bump'
   spec.add_development_dependency 'single_cov'
   spec.add_development_dependency 'rubocop'
   spec.add_development_dependency 'mocha'
   spec.add_development_dependency 'forking_test_runner'
-  spec.add_runtime_dependency 'actionpack', '>= 4.2', '< 6.2'
+  spec.add_runtime_dependency 'actionpack', '>= 4.2'
```
Comment (grosser) [inline stronger_parameters.gemspec:23]:
```suggestion
  spec.add_runtime_dependency 'actionpack', '>= 4.2', '< 7.1'
```

File: test/controller_test.rb [hunk 2]
```diff
@@ -24,6 +24,8 @@ def create
   def post(action, options = {})
```
Comment (grosser) [inline test/controller_test.rb:24]: this is changing the method signature, so I think the if should be around different  `def`s

Comment (grosser) [reply test/controller_test.rb:24]: ... but I guess does not matter since it's just for our tests

File: test/test_helper.rb [hunk 3]
```diff
@@ -6,10 +6,14 @@
 require 'single_cov'
 SingleCov.setup :minitest
 
+require 'rails'
 require 'maxitest/global_must'
 require 'maxitest/autorun'
-require 'mocha/setup'
-require 'rails'
+if Rails::VERSION::MAJOR >= 7
+  require 'mocha/minitest'
```
Comment (grosser) [inline test/test_helper.rb:13]: this should already work on older versions

**Comment by grosser** [comment]
might be best to not bundle update the old gemfiles locks, they should be on their way out soon anyways
... also maybe start by adding new ruby versions and then rails as it's own PR to isolate issues

**Comment by grosser** [comment]
bumping ruby here https://github.com/zendesk/stronger_parameters/pull/94 🤞

File: test/test_helper.rb (hunk 3 above)
Comment (grosser) [reply test/test_helper.rb:13]: ... maybe just keep it for now, otherwise we have to update the old lockfiles and that might cause even more chaos

**Comment by grosser** [comment]
updating ruby versions failed 😞 
... they only work with new rails versions

also this style might make the actions more readable:
```
        - {ruby: '2.6', rails: '4.2', task: test}
        - {ruby: '2.7', rails: '4.2', task: test}
        - {ruby: '3.0', rails: '4.2', task: test}
        - {ruby: '3.1', rails: '4.2', task: test}
```

**Comment by skateman** [comment]
@Nkadze are you willing to address @grosser's comments? If not, I can take this over as we need this.

Also saying hello to @vecerek :laughing: :wink:

**Comment by Nkadze** [comment]
Hi Guys, sorry for being inactive for a while, I have very busy schedule last weeks.
@skateman sure, you can take over, thanks for helping out

**Pull request closed without merging**